doc/examples.md

```diff
@@ -1,39 +1,139 @@
+
+The Functional API provides access to individual oneDNN primitives.
+
+### Fundamental Concepts and API Basics
+
+- @ref getting_started_cpp - Basic API workflow and primitive execution
+- @ref memory_format_propagation_cpp - Memory format optimization and propagation patterns
+- @ref cross_engine_reorder_cpp - Data movement between CPU and GPU engines
+
+### Interoperability with External Runtimes
+
+- @ref sycl_interop_buffer_cpp - SYCL with buffer-based memory management
+- @ref sycl_interop_usm_cpp - SYCL with Unified Shared Memory
+- @ref gpu_opencl_interop_cpp - OpenCL interoperability API
+
+### Matrix Multiplication with Different oneDNN Features
+
+Basic Operations:
+- @ref matmul_example_cpp - Basic MatMul primitive usage
+- @ref cpu_sgemm_and_matmul_cpp - SGEMM vs MatMul APIs comparison on CPU
+
+Quantization flavors:
+- @ref matmul_f8_quantization_cpp - f8 quantization
+- @ref cpu_matmul_quantization_cpp - CPU-optimized int8 quantization
+- @ref inference_int8_matmul_cpp - int8 inference optimization techniques
+
+Advanced Usages:
+- @ref matmul_with_host_scalar_scale_cpp - MatMul with scaling factor on the host
+- @ref cpu_matmul_coo_cpp - CPU Sparse MatMul in Coordinate (COO) format
+- @ref cpu_matmul_csr_cpp - CPU Sparse MatMul in Compressed Sparse Row (CSR) format
+- @ref cpu_matmul_weights_compression_cpp - CPU weights compression
+- @ref weights_decompression_matmul_cpp - Weight decompression workflows
+
+### Inference and Training
+
+Neural network implementations demonstrating inference and training workflows:
+
+| Type | Precision | Mode | Example | Key Features |
+|:-------------|:----------|:-----|:--------|:-------------|
+| CNN | f32 | Inference | @ref cnn_inference_f32_cpp | Basic floating-point inference |
+| CNN | int8 | Inference | @ref cnn_inference_int8_cpp | Quantized inference |
+| CNN | f32 | Training | @ref cnn_training_f32_cpp | Forward and backward passes |
+| CNN | bf16 | Training | @ref cnn_training_bf16_cpp | Mixed-precision training |
+| RNN | f32 | Inference | @ref cpu_rnn_inference_f32_cpp | RNN inference on CPU |
+| RNN | int8 | Inference | @ref cpu_rnn_inference_int8_cpp | Quantized RNN inference on CPU |
+| RNN | f32 | Training | @ref rnn_training_f32_cpp | RNN training |
+
+### Recurrent Neural Networks
+
+- @ref vanilla_rnn_example_cpp - Basic RNN: Fundamental recurrent cell implementation
+- @ref lstm_example_cpp - LSTM: Long Short-Term Memory with gating mechanisms
+- @ref lbr_gru_example_cpp - LBR-GRU: Linear-Before-Reset Gated Recurrent Unit
+- @ref augru_example_cpp - AU-GRU: Attention-Update-Gate RNN variant
+
+### Performance Analysis
+
+A few techniques for performance measurements:
+- @ref performance_profiling_cpp - Performance analysis based on `ONEDNN_VERBOSE` usage
+- @ref matmul_perf_cpp - Simple Matrix multiplication performance testing
+
+### Microkernel Interface (uKernel)
+
+- @ref cpu_brgemm_example_cpp - Batch-Reduced GEMM (BRGEMM) microkernel implementation on CPU
```

Comment:
Maybe move this into it's own top level section?

Reply:
ping @vpirogov :)

Reply:
This structure makes sense. One small thing is that we are positioning microkernel API as an entity as the same level as Functional and Graph.

Reply:
Sure, updated